renderer/pages/dashboard.tsx

```diff
@@ -0,0 +1,166 @@
+import React from "react";
+import { MainLayout } from "../layouts";
+import parsePrometheusTextFormat from 'parse-prometheus-text-format';
+import { Arweave_API } from "../lib/arweave";
+import { setMinorState, selectMinorState } from "../store/minorSlice";
+import { useDispatch, useSelector } from "react-redux";
+import { MinorParser } from "../types/Minor";
+
+interface MenuItems {
+    label: string;
+    target: string;
+    subMenuItems?: SubMenuItems[];
+}
+interface SubMenuItems {
+    label: string;
+    target: string;
+}
+
+export default function DashboardPage() {
+    const minorState = useSelector(selectMinorState);
+    const dispatch = useDispatch();
+
+    const [data, setData] = React.useState(null);
+    const menuItems: MenuItems[] = [
+        {
+            label: 'Core',
+            target: 'section-1',
+            subMenuItems: [
+                { label: 'Data Related', target: 'sub-section-1-1' },
+                { label: 'Hash Rate', target: 'sub-section-1-2' },
+                { label: 'Earnings', target: 'sub-section-1-3' },
+            ],
+        },
+        {
+            label: 'Advanced',
+            target: 'section-2',
+            subMenuItems: [
+                { label: 'Performance', target: 'sub-section-2-1' },
+                { label: 'Debug', target: 'sub-section-2-2' },
+                { label: 'Raw Logs', target: 'sub-section-2-3' },
+            ],
+        },
+    ];
+
+    const [activeMenuItem, setActiveMenuItem] = React.useState<string>(menuItems[0].target);
+
+    const handleScroll = () => {
+        const scrollPosition = window.scrollY;
+        const sectionOffsets: any[] = menuItems.flatMap((item: MenuItems) => [
+            document.getElementById(item.target).offsetTop - 100,
+            ...(item.subMenuItems || []).map((subItem: SubMenuItems) => document.getElementById(subItem.target)?.offsetTop - 100),
+        ]);
+
+        const tempMenuList = menuItems.flatMap((item: MenuItems) => [
+            item,
+            ...(item.subMenuItems || []).map((subItem: SubMenuItems) => subItem),
+        ]);
+
+        const activeIndex: number = sectionOffsets.findIndex(offset => offset > scrollPosition);
+        setActiveMenuItem(() => {
+            const activeMenu = tempMenuList[activeIndex > tempMenuList.length ? activeIndex - 1 : activeIndex];
+            return activeMenu.target || activeMenu[0].target;
+        });
+    };
+
+    React.useEffect(() => {
+        window.addEventListener('scroll', handleScroll);
```

Comment:
I'd use a library for this, because of timing issues (and potential errors) related to window references